features/FEATURE_BLE/targets/TARGET_Maxim/MaximBLE.cpp

```diff
@@ -52,29 +52,38 @@
 #include "hci_vs.h"
 
 /* Number of WSF buffer pools */
-#define WSF_BUF_POOLS               4
+#define WSF_BUF_POOLS 5
 
 /*! Free memory for pool buffers. */
-static uint8_t mainBufMem[768];
+static uint8_t mainBufMem[1040];
 
 /*! Default pool descriptor. */
 static wsfBufPoolDesc_t mainPoolDesc[WSF_BUF_POOLS] =
 {
   {  16,  8 },
   {  32,  4 },
   {  64,  2 },
-  { 128,  2 }
+  { 128,  2 },
+  { 272,  1 }
 };
 
 /*! WSF handler ID */
 wsfHandlerId_t maximHandlerId;
 static volatile int reset_complete;
 
+#ifdef BLE_HCI_UART
```

Comment:
Currently there are two Maxim HCI drivers; one for SPI and the other UART. These are separate files and do not require the use of ifdef's.  Until something changes the HCI driver initialization is the common point within MaximBLE.cpp and thus needs to know what type of mbed objects are needed to reserve peripherals for HCI use. 

The HCI drivers use HAL pin mapping to determine the peripheral instance associated with the given SPI slave select or UART transmit pin.  With the peripheral instance the other peripheral pins can be determined programmatically.  The other pin arguments represent a single pin that can be used without issue. So yes the pins can be mapped to other pins with the same functionality, although this has not been tested.